### PR TITLE
Adopt identical-to and further clarifications for both WGSL and WebGPU

### PR DESCRIPTION
**Pull request by mehmetoguzderin**
This PR does not alter any behavior and introduces (beneficially) verbose wording and Infra reuse to help anticipated i18n https://github.com/w3c/i18n-activity/issues/1501 and https://github.com/w3c/i18n-activity/issues/1508 and https://github.com/w3c/i18n-activity/issues/1509 and https://github.com/w3c/i18n-activity/issues/1510 feedback.

***
<a href="https://pr-preview.s3.amazonaws.com/mehmetoguzderin/gpuweb/pull/2767.html" title="Last updated on Apr 20, 2022, 7:54 PM UTC (4c4c20a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2767/cbfa42b...mehmetoguzderin:4c4c20a.html" title="Last updated on Apr 20, 2022, 7:54 PM UTC (4c4c20a)">Diff</a>

### CODE REVIEW

File: spec/index.bs
Comment (toji) [inline spec/index.bs:4708]: The above spec prose and associated seem to be copied verbatim in a few places. Could we establish a definition for something like "WGSL identifier comparison rules.", give this definition there, then link to it everywhere else with text along the lines of:

`WGSL names (identifiers) in source maps follow [=WGSL identifier comparison rules=].`

**Review by toji — COMMENTED** [review]
API side changes seems fine, though I'll defer to @dneto0 for a final approval of the text.

One non-blocking suggestion regarding consolidating this prose to a common definition.

File: spec/index.bs
Comment (mehmetoguzderin) [reply spec/index.bs:4708]: @toji Thank you for the suggestion; that makes sense to me. If other editors agree to text wording changes, I will append a commit to form such and deduplicate in this PR!

Comment (mehmetoguzderin) [reply spec/index.bs]: @toji I hope the latest commit's deduplication resolves this without breaking the existing opinion!

**Review by toji — APPROVED** [review]
Thanks! This feels quite a bit cleaner to me, and allows the WGSL spec to "own" the logic for comparing WGSL identifiers.

I'd still request that @dneto0 OK the language used before landing, but I'm happy with the formatting!

File: wgsl/index.bs
Comment (dneto0) [inline wgsl/index.bs:647]: This is subtly different from the text it is replacing.  It's a strong statement in fact.

Consider this text:

         fn foo() { let  H = ... ; }
         fn bar() { let H = ...; }

Assume the two `H` have distinct code points but are homoglyphs for each other.  The scopes of those identifers don't overlap.   So the program doesn't change meaning if you replace instances of one with its homoglyph.


Suggest:

> Note:
A user agent should issue developer-visible warnings when the meaning of a WGSL program would change if all instances of an identifier are replaced with one of that identifier's homographs.
(A homoglyph is a sequence of code points that may appear the same to a reader as another sequence of code points. 
Examples of mappings to detect homoglyphs are the transformations, mappings, and matching algorithms mentioned in the previous paragraph.
Two sequences of code points are homographs if you can transform one into the other by repeatedly replacing a subsequence with its homoglyph.)

**Review by dneto0 — CHANGES_REQUESTED** [review]
Boy, I learned a lot.

I have some requests for changes.

File: wgsl/index.bs
Comment (mehmetoguzderin) [reply wgsl/index.bs:641]: @dneto0 Thank you, I think this change conveys the idea much better, and it is helpful that we have the history of intentional change to spell out what we do exactly. I made the first sentence not have a passive voice as a slight difference. Is that OK?

File: wgsl/index.bs
Comment (mehmetoguzderin) [reply wgsl/index.bs]: @dneto0 Done, this is much better!

**Review by dneto0 — APPROVED** [review]
Looks great.
Thanks!

**Comment by dneto0** [comment]
I see @toji has approved the API side of the change, so I will land this.

**Pull request merged**
